platform/darwin/src/MGLStyle.mm

```diff
@@ -85,6 +85,10 @@ @interface MGLStyle()
 @property (readonly, copy, nullable) NSURL *URL;
 @property (nonatomic) NS_MUTABLE_DICTIONARY_OF(NSString *, NS_DICTIONARY_OF(NSObject *, MGLTextLanguage *) *) *localizedLayersByIdentifier;
 
+// Used for retain/release management
+@property (nonatomic) NSMutableSet *layersForUpdating;
```

Comment:
Does the use of these properties remove the need for tracking the peer object in core for style layers? For layers created in the SDK, `[MGLStlye layerFromMBGLLayer]` could be used to lookup this set instead of querying core